x-pack/plugins/reporting/export_types/printable_pdf/server/lib/screenshots.js

```diff
@@ -112,7 +112,11 @@ export function screenshotsObservableFactory(server) {
 
         function waitForRender(visualization) {
           return new Promise(function (resolve) {
-            visualization.addEventListener('renderComplete', () => resolve());
+            // The renderComplete fires before the visualizations are in the DOM, so
+            // we wait for the event loop to flush before resolving the promise. This
+            // seems to correct the timing issue and prevents us from capturing
+            // visualization screenshots before they're rendered.
+            visualization.addEventListener('renderComplete', () => setTimeout(resolve));
```

Comment:
We could probably put the setTimeout in a `.then` at the bottom of this function... (e.g. `return Promise.all(renderedTasks).then(new Promise(r => setTimeout(r)));` That would prevent us from queuing up a setTimeout per visualization, but would still give us the desired effect of waiting for the event queue to flush before resolving.